Formula/g/glibc@2.17.rb

```diff
@@ -0,0 +1,190 @@
+    OS.kernel_version >= MINIMUM_LINUX_KERNEL_VERSION
+  end
+
+  def message
+    <<~EOS
+      Linux kernel version #{MINIMUM_LINUX_KERNEL_VERSION} or later is required by glibc.
+      Your system has Linux kernel version #{OS.kernel_version}.
+    EOS
+  end
+
+  def display_s
+    "Linux kernel #{MINIMUM_LINUX_KERNEL_VERSION} (or later)"
+  end
+end
+
+class GlibcAT217 < Formula
+  desc "GNU C Library"
+  homepage "https://www.gnu.org/software/libc/"
+  url "https://ftp.gnu.org/gnu/glibc/glibc-2.17.tar.gz"
+  mirror "https://ftpmirror.gnu.org/gnu/glibc/glibc-2.17.tar.gz"
+  sha256 "a3b2086d5414e602b4b3d5a8792213feb3be664ffc1efe783a829818d3fca37a"
+  license all_of: ["GPL-2.0-or-later", "LGPL-2.1-or-later"]
+
+  keg_only :versioned_formula
+
+  depends_on GawkRequirement => :build
+  depends_on "linux-headers@4.4" => :build
+  depends_on MakeRequirement => :build
+  depends_on SedRequirement => :build
+  depends_on :linux
+  depends_on LinuxKernelRequirement
+
+  # Backport of:
+  # https://sourceware.org/git/?p=glibc.git;a=commit;h=e9177fba13549a8e2a6232f46080e5c6d3e467b1
+  patch do
+    url "https://git.centos.org/rpms/glibc/raw/ca483cc5b0e3e6a595a2c103755dee4d72f14f25/f/SOURCES/glibc-rh1500908.patch"
+    sha256 "48bfb15f5a26161bbef3d58e18d44876a170ddbfcc5922a39c77fd8da9fe68f6"
+  end
+
+  # Backport of:
+  # https://sourceware.org/git/?p=glibc.git;a=commit;h=43d06ed218fc8be58987bdfd00e21e5720f0b862
+  patch do
+    url "https://raw.githubusercontent.com/Homebrew/formula-patches/edf27e2c185b2914a864c66d8fcdca966d73b407/glibc/2.17-aarch64-dl-machine.diff"
+    sha256 "ece66819f9ef3a1b73081d0f5cda8c814b5204d25c07f8a7adb8209ee286c39d"
+  end
+
+  def install
+    # Fix checking version of gnumake... 4.3, bad
+    inreplace "configure",
+              "3.79* | 3.[89]*)",
+              "3.79* | 3.[89]* | [4-9].* | [1-9][0-9]*)"
+
+    # Fix checking version of gcc-11... 11.4.0, bad
+    inreplace "configure",
+              "4.[3-9].* | 4.[1-9][0-9].* | [5-9].* )",
+              "4.[3-9].* | 4.[1-9][0-9].* | [5-9].* | [1-9][0-9]* )"
+
+    # Setting RPATH breaks glibc.
+    %w[
+      LDFLAGS LD_LIBRARY_PATH LD_RUN_PATH LIBRARY_PATH
+      HOMEBREW_DYNAMIC_LINKER HOMEBREW_LIBRARY_PATHS HOMEBREW_RPATH_PATHS
+    ].each { |x| ENV.delete x }
+
+    # Fix relocation R_X86_64_32S against symbol `__libc_csu_fini' can not be
+    # used when making a PIE object; recompile with -fPIE
+    # See https://sourceware.org/pipermail/libc-alpha/2020-March/111688.html
+    ENV.append "LDFLAGS", "-no-pie" if Hardware::CPU.intel?
+
+    # Use brewed ld.so.preload rather than the host's /etc/ld.so.preload
+    inreplace "elf/rtld.c",
+              '= "/etc/ld.so.preload";',
+              '= SYSCONFDIR "/ld.so.preload";'
+
+    mkdir "build" do
+      args = [
+        "--disable-debug",
+        "--disable-dependency-tracking",
+        "--disable-silent-rules",
+        "--prefix=#{prefix}",
+        "--sysconfdir=#{etc}",
+        "--enable-obsolete-rpc",
```

Comment:
I wonder if we can drop this flag

Reply:
Yes, probably. https://github.com/Linuxbrew/legacy-linuxbrew/pull/603 doesn't seem relevant anymore.

Reply:
Yeah we'll probably drop it from regular glibc too but for now dropping it from here is safe as it's only used in Portable Ruby and it shouldn't be using RPC AFAIK.

Reply:
Dropped in the latest push.

Actually, `--enable-obsolete-rpc` seems safe to remove from both `glibc` 2.35 and `glibc@2.13`. This option was introduced in 2.16 (https://github.com/bminor/glibc/commit/021db4be6f1f4189f66feee066a495d49e92b93e) and removed in 2.32 (https://github.com/bminor/glibc/commit/5500cdba4018ddbda7909bc7f4f9718610b43cf0) so both 2.35 and 2.13 happen to be outside the range!

Reply:
- https://github.com/Homebrew/homebrew-core/pull/198033
- https://github.com/Homebrew/homebrew-core/pull/198034